### PR TITLE
Clean up configs for single source of truth for tokens 

### PR DESCRIPTION
**Pull request by johnmosesman**
Overall changes:

* The config is now the source of token of what the valid tokens for that network is.
* This simplifies the token lookup/market building, and we don't have to keep track of valid tokens in multiple places.
* The one downside here is we lose the ability to have the keys typed as `TokenName`, but this is what is causing the double-entry of configs.

TODO:

* Move price oracle address onto token, will help clean up a lot of piping things down

TODO not in this PR:
* Removed all the other configs except Rinkeby, and will add a metis testnet one (they were all garbage addresses anyways). Our loading state for non-supported network likely needs to be better.

### CODE REVIEW

File: app/hooks/use-markets.ts [hunk 1]
```diff
@@ -99,7 +98,6 @@ export function useMarkets(
       return {
         id: tp.token.symbol,
         tokenPair: tp,
-        tokenMetaData: tokenMetaData[tp.token.symbol as TokenName],
```
Comment (johnmosesman) [inline app/hooks/use-markets.ts:102]: `tokenMetadata` is redundant now that all the config properties live on the token

File: app/types/global.ts [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
-export interface cToken {
+export interface Token {
+  symbol: string;
+  icon: string;
   name: string;
-  address: string;
   decimals: number;
+  address: string;
+  cToken: cToken;
 }
```
Comment (johnmosesman) [inline app/types/global.ts:15]: Token now has all the information it needs for that token, TODO: add price oracle.

File: app/hooks/use-token-pairs.ts [hunk 3]
```diff
@@ -7,18 +7,17 @@ function generateTokenPairs(
...
-        token: networkData.Tokens[tokenMetaDatum.symbol],
-        cToken: networkData.cTokens[tokenMetaDatum.cTokenSymbol],
-      };
-    });
+  return tokenSymbols.map((symbol: string): TokenPair => {
+    const tokenMetaDatum = networkData.Tokens[symbol];
+
+    return {
+      token: networkData.Tokens[tokenMetaDatum.symbol],
+      cToken: networkData.cTokens[tokenMetaDatum.cTokenSymbol],
```
[the remark this answers is not in the document]
Comment (johnmosesman) [reply app/hooks/use-token-pairs.ts]: cToken is everywhere in the codebase so if we want that it'd be a bigger change

File: app/types/global.ts [hunk 4]
```diff
@@ -1,17 +1,19 @@
 import type { TokenMetaDatum } from "~/config/tokenMetaData";
 
 export interface Token {
-  symbol: TokenName;
+  symbol: string;
   icon: string;
   name: string;
   decimals: number;
   address: string;
+  cTokenSymbol: string;
```
[the remark this answers is not in the document]
Comment (johnmosesman) [reply app/types/global.ts]: see ^

File: app/hooks/use-markets.ts (hunk 1 above)
Comment (ryanlabouve) [reply app/hooks/use-markets.ts:102]: nice

File: app/hooks/use-on-supported-network.ts [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import { useState, useEffect } from "react";
 
 // Reference ~/connectors/index.ts for chain mappings
+// TODO: Are all these still relevant after nuking the bad configs?
```
Comment (ryanlabouve) [inline app/hooks/use-on-supported-network.ts:4]: We should add chainId to the config and then reference that for supported chains

File: app/hooks/use-tender-context.ts [hunk 6]
```diff
@@ -1,40 +1,17 @@
 import { useState, useEffect } from "react";
-import { type Market, type TenderContext, TokenName } from "~/types/global";
+import { type Market, type TenderContext } from "~/types/global";
 import { hooks as Web3Hooks } from "~/connectors/meta-mask";
 import { useTokenPairs } from "./use-token-pairs";
 import { useOnSupportedNetwork } from "./use-on-supported-network";
 import { useNetworkData } from "./use-network-data";
 import { useMarkets } from "./use-markets";
 
-// TODO: Move to config
```
Comment (ryanlabouve) [inline app/hooks/use-tender-context.ts:9]: 🇳🇮

Comment (ryanlabouve) [reply app/hooks/use-tender-context.ts:9]: Tried for `:nice:` that 🇳🇮 works too.

File: app/hooks/use-token-pairs.ts [hunk 7]
```diff
@@ -1,29 +1,42 @@
...
-  networkData: NetworkData,
-  supportedTokenNames: TokenName[]
-): TokenPair[] {
-  return supportedTokenNames
-    .filter((tokenName) => {
-      return networkData.Tokens[tokenName];
-    })
-    .map((tokenName: TokenName): TokenPair => {
-      const tokenMetaDatum = tokenMetaData[tokenName];
+  let tokenPairs: (TokenPair | null)[] = tokenSymbols.map(
```
Comment (ryanlabouve) [inline app/hooks/use-token-pairs.ts:7]: Line 7 and 8 are hard to read

Comment (ryanlabouve) [reply app/hooks/use-token-pairs.ts:7]: `[null, null, null]` is probably not good

Comment (ryanlabouve) [reply app/hooks/use-token-pairs.ts:7]: ~good~ *not the type you are intending

**Pull request merged**